### PR TITLE
Enhancement: CalciteAction text modes

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** Enhancement: CalciteAction text modes #313

## Summary

Enhancement: CalciteAction text display modes

#### Desired New Behavior

API Proposal:

```ts
interface CalciteAction {
  @Prop({reflect:true}) textDisplay: "hidden"  | "visible" | "interactive";
  @Prop({reflect:true}) textEnabled: boolean; // deprecate!!
}
```

#### Desired Change to Look and Feel

- "hidden" will never show the text.
- "visible" will always show the text.
- "interactive" will show the text when element is hovered or focused.

```html
<calcite-action text="hello world" text-display="interactive" />
```

### CODE REVIEW

**Comment by driskull** [comment]
Will create a seperate issue for auto text styling for @asangma  once merged.

File: src/components/calcite-action-bar/calcite-action-bar.tsx [hunk 1]
```diff
@@ -103,11 +103,13 @@ export class CalciteActionBar {
     return shellNode.layout;
   }
 
-  setTextEnabled(expanded: boolean): void {
+  setTextMode(expanded: boolean): void {
```
Comment (jcfranco) [inline src/components/calcite-action-bar/calcite-action-bar.tsx]: [Nitpick] `setActionTextMode`?

File: src/components/calcite-action/calcite-action.tsx [hunk 2]
```diff
@@ -51,10 +51,15 @@ export class CalciteAction {
...
+  /**
+   * @deprecated Use 'textMode' instead.
+   */
+  @Prop() textEnabled = false;
+
   /**
    * Indicates whether the text is displayed.
    */
-  @Prop({ reflect: true }) textEnabled = false;
+  @Prop({ reflect: true }) textMode: "show" | "hide" | "auto" = "hide";
```
Comment (jcfranco) [inline src/components/calcite-action/calcite-action.tsx]: Modes should probably be moved to a type.

**Comment by jcfranco** [comment]
> all of our textProp properties are for setting text strings which is inconsistent with this name.

This was already the case w/ `textEnabled`, so not a showstopper IMO.

File: src/components/calcite-action/calcite-action.tsx (hunk 2 above)
Comment (jcfranco) [reply src/components/calcite-action/calcite-action.tsx]: WDYT about `textDisplay = "visible" | "hidden" | "on-interaction"`? I'm sure there's a better word for that last one. 😄

Comment (driskull) [reply src/components/calcite-action/calcite-action.tsx]: @jcfranco Yeah I like `visible`, `hidden` much better.

How about just `interactive`?

**Comment by driskull** [comment]
Changed to `interactive` for now.

Could also be

- `textMode="animated"`

**Pull request merged**